Remove hardcoded certificates from Kubernetes installation files (#54)
strimzi/drain-cleaner: packaging/install/kubernetes/README.md

So the web-hook service needs to have HTTPS support.
And the CA of the certificate used for this service needs to be specified in the `ValidatingWebhookConfiguration` resource.

This directory contains sample installation files without the certificates.

> You can generate the certificates by following the guide below and update the installation files with the,

Reviewer: ```suggestion
The following procedures describe how to generate certificates and add them to installation files.
```

> ## Generating the certificate using OpenSSL

Reviewer: ```suggestion
## Generating certificates using OpenSSL
```

> You can use the OpenSSL utility to generate the TLS certificate for the Strimzi Drain Cleaner webhook.

Reviewer: ```suggestion
Use the OpenSSL TLS management tool to generate the TLS certificate for the Strimzi Drain Cleaner webhook.
```

> The steps below have been tested with OpenSSL 1.1.1 and should work on Linux, MacOS or in the Windows Subsystem for Linux.

Reviewer: ```suggestion
The steps below have been tested with OpenSSL 1.1.1 and should work on Linux, MacOS, or in the Windows Subsystem for Linux.
```

> 1) Create a subdirectory `tls-certificate` where we will generate the certificate files and enter it:

Reviewer: ```suggestion
1) Create and navigate to a subdirectory called `tls-certificate`:
```

   ```
   mkdir tls-certificate
   cd tls-certificate
   ```

> 2) Generate the CA public and private keys:

Reviewer: ```suggestion
2) Generate a CA public certificate and private key in the  `tls-certificate` directory:
```

>    ```
>    openssl req -nodes -new -x509 -keyout ca.key -out ca.crt -subj "/CN=Strimzi Drain Cleaner CA"
>    ```

Reviewer: ```suggestion 
    ```
    openssl req -nodes -new -x509 -keyout ca.key -out ca.crt -subj "/CN=Strimzi Drain Cleaner CA"
    ```
    
   A `ca.crt` and `ca.key` file is created. 
```

> 3) Generate the private key for the Strimzi Drain Cleaner:

Reviewer: ```suggestion
3) Generate the private TLS key for the Strimzi Drain Cleaner:
```

>    ```
>    openssl genrsa -out tls.key 2048
>    ```

Reviewer: ```suggestion
    ```
    openssl genrsa -out tls.key 2048
    ```
    
    A `tls.key` file is created.
```

Reviewer: If you make the changes, check the formatting on the last couple of suggestions. They weren't working as expected in the comments boxes

> 4) Generate the Certificate Signing Request and sign it with the CA from step 1:

Reviewer: ```suggestion
4) Generate a Certificate Signing Request and sign it by adding the CA public certificate (`ca.crt`) you generated:
```

   ```
   openssl req -new -key tls.key -subj "/CN=strimzi-drain-cleaner.strimzi-drain-cleaner.svc" \
        | openssl x509 -req -CA ca.crt -CAkey ca.key -CAcreateserial -extfile <(printf "subjectAltName=DNS:strimzi-drain-cleaner.strimzi-drain-cleaner.svc") -out tls.crt

>    ```

Reviewer: ```suggestion
    ```
    A `tls.crt` file is created.
```

   If you plan to change the name of the Strimzi Drain Cleaner service or install it into a different namespace, you have to change the common name (CN) of the certificate.

>    The CN should follow the pattern `<SERVICE_NAME>.<NAMESPACE_NAME>.svc`.

Reviewer: ```suggestion
   The CN must follow the pattern `<SERVICE_NAME>.<NAMESPACE_NAME>.svc`.
```

5) The `tls-certificate` directory should now contain several certificate files which we will use in the installation files.
   You can exit the `tls-certificate` directory now.
   ```
   cd ..

>    ```

Reviewer: I'm not sure we need step 5?

## Updating the installation files with the generated certificates

> This section expects that you have the following certificate files generated in the previous section.
> If you generated your certificates in a different way or on a different path, you should update the path in the commands.

Reviewer: ```suggestion
After you have generated the certificates you need, update the installation files.
This procedure assumes that you used the previous procedure to generate the certificate files.
If you generated your certificates in a different way or on a different path, you should update the path in the commands.
```

> 1) Edit the [`070-ValidatingWebhookConfiguration.yaml`](070-ValidatingWebhookConfiguration.yaml) file and update the `caBundle` field to contain the Base64 encoded public key of your CA.

Reviewer: ```suggestion
1) Edit the `caBundle` field in the [`070-ValidatingWebhookConfiguration.yaml`](070-ValidatingWebhookConfiguration.yaml) installation file to specify a Base64 encoded public key of your CA.
```

>    You can get the Base64 encoded public key for example using the `base64` utility:

Reviewer: ```suggestion
   You can use the `base64` utility to get the Base64 encoded public key:
```

   ```
   base64 tls-certificate/ca.crt
   ```

>    Use the output of one of these commands as the `caBundle`.
>    You YAML should look similar to this:

Reviewer: ```suggestion
   After adding the Base64 encoded public key, your YAML should look similar to this:
```

    # ...
   ```
2) Create the `strimzi-drain-cleaner` namespace:
   ```
   kubectl create ns strimzi-drain-cleaner
   ```

> 3) Create the Secret named `strimzi-drain-cleaner` with the `tls.crt` and `tls.key` files.
>    This Secret will be used by the Strimzi Drain Cleaner deployment.
>    You can create the Secret using `kubectl`:
>    ```
>    kubectl create secret tls strimzi-drain-cleaner \
>        -n strimzi-drain-cleaner  \
>        --cert=tls-certificate/tls.crt \
>        --key=tls-certificate/tls.key
>    ```
>    The resulting Secret should look similar to this:

Reviewer: ```suggestion
3) Use `kubectl` to create a secret named `strimzi-drain-cleaner` with the `tls.crt` and `tls.key` files you generated:
   
     ```
   kubectl create secret tls strimzi-drain-cleaner \
       -n strimzi-drain-cleaner  \
       --cert=tls-certificate/tls.crt \
       --key=tls-certificate/tls.key
    ```
   This secret is used by the Strimzi Drain Cleaner deployment.
   The resulting Secret should look similar to this:
```
